contracts/interfaces/IBonus.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+
+interface IBonus {
+    /**
+     * @notice Info of each user.
+     * `xp` The amount of XP.
+     * `allocation` User allocation
+     * `discountPresale` Discount during presale
+    **/
+    struct UserInfo {
+        uint256 xp;
+        uint256 allocation;
+        uint256 discountPresale;
+    }
+
+    /**
+     * @notice Info of each admin.
+     * `xpLimit` The amount of XP that admin can distribute over other users
+     * `levelLimit` Admins can't distribute tokens so that (user level > levelLimit)
+    **/
+    struct AdminInfo {
```

Comment:
Admin:
- Can setup communityAdmin

CommunityAdmin:
- Can distribute XP

User:
- Can mint NFT according to the level